indico/modules/rb/client/js/components/TimeRangePicker.jsx

```diff
@@ -9,62 +9,62 @@ import _ from 'lodash';
 import moment from 'moment';
 import PropTypes from 'prop-types';
 import React from 'react';
-import {Dropdown} from 'semantic-ui-react';
 
+import {ComboBox} from 'indico/react/components';
 import {FinalField} from 'indico/react/forms';
-import {Translate} from 'indico/react/i18n';
+import {PluralTranslate, Translate} from 'indico/react/i18n';
 import {serializeTime, toMoment} from 'indico/utils/date';
 
 import './TimeRangePicker.module.scss';
 
-const ARROW_KEYS = ['ArrowUp', 'ArrowDown', 'Up', 'Down'];
+// FIXME: Alignment of time options in the list
+
 const START_HOUR = '06:00';
+const ALLOWED_INPUT_FORMATS = ['HH:mm', 'HHmm', 'H:mm', 'Hmm'];
+const MIN_DURATION = moment.duration(30, 'minutes');
 
 function _humanizeDuration(duration) {
-  const hours = duration.hours();
   const minutes = duration.minutes();
-  if (hours === 1 && minutes === 0) {
-    return Translate.string('1 hour');
-  } else if (hours !== 0) {
-    return Translate.string('{time} hours', {time: hours + minutes / 60});
-  } else {
-    return Translate.string('{time} min', {time: minutes});
+  const hours = duration.hours() + minutes / 60;
+  if (hours < 1) {
+    return PluralTranslate.string('{time} min', '{time} min', minutes, {time: minutes});
   }
+  return PluralTranslate.string('{time} hour', '{time} hours', hours, {time: hours});
 }
 
 function generateStartTimeOptions(minTime) {
```

Comment:
![image](https://github.com/user-attachments/assets/f24d1bd6-6bf5-4de4-a3d8-8a72efd14c38)

Looks like it's taken the current minute into account when generating the dropdown...